File: app/types/manutencao.ts
export type StatusManutencao = 'Agendadas' | 'Concluidas' | 'Canceladas' | 'Urgentes' | 'Em Andamento';

export interface Manutencao {
    id: number;
    equipamentoId: number;
    equipamentoNome?: string;
    tipoManu: string
    descricao: string;
    dataAbertura: string;
    dataFechamento: string;
    valor: number;
    status: StatusManutencao;
}

File: api/manutencao/manutencao.go
package manutencao

import (
	"time"

	"go.mongodb.org/mongo-driver/v2/bson"
)

type Manutencao struct {
	ID             bson.ObjectID `bson:"_id" json:"id"`
	EquipamentoID  string        `bson:"equipamentoId" json:"equipamentoId"`
	TipoManu       string        `bson:"tipoManu" json:"tipoManu"`
	Descricao      string        `bson:"descricao" json:"descricao"`
	DataAbertura   time.Time     `bson:"dataAbertura" json:"dataAbertura"`
	DataFechamento time.Time     `bson:"dataFechamento" json:"dataFechamento"`
	Valor          float64       `bson:"valor" json:"valor"`
	Status         string        `bson:"status" json:"status"`
}
